test/functional/feature_segwit.py

```diff
@@ -279,6 +280,34 @@ def run_test(self):
         # Mine a block to clear the gbt cache again.
         self.nodes[0].generate(1)
 
+        self.log.info("Signing with all-segwit inputs reveals fee rate")
+        # Test that signing a tx with all-segwit inputs returns the fee
+        # in the results
+        addr = self.nodes[0].getnewaddress(address_type='p2sh-segwit')
+        txid = self.nodes[0].sendtoaddress(addr, 1)
+        tx = self.nodes[0].getrawtransaction(txid, True)
+        n = -1
+        value = -1
+        for o in tx["vout"]:
+            if o["scriptPubKey"]["addresses"][0] == addr:
+                n = o["n"]
+                value = Decimal(o["value"])
+                break
+        assert n > -1 # failure means we could not find the address in the outputs despite sending to it
+        assert_approx(value, 1.0, 0.01) # failure means we got an unexpected amount of coins, despite trying to send 1
+        fee = Decimal("0.00010000")
+        value_out = value - fee
+        self.nodes[0].generatetoaddress(1, self.nodes[0].getnewaddress())
+        raw = self.nodes[0].createrawtransaction([{"txid" : txid, "vout" : n}], [{self.nodes[0].getnewaddress() : value_out}])
+        signed = self.nodes[0].signrawtransactionwithwallet(raw)
+        assert_equal(signed["complete"], True)
+        txsize = self.nodes[0].decoderawtransaction(signed['hex'])['vsize']
+        exp_feerate = 1000 * fee / Decimal(txsize)
+        assert_approx(signed["feerate"], exp_feerate, Decimal("0.00000010"))
+        # discrepancy = 100000000 * (exp_feerate - signed["feerate"])
+        # assert -10 < discrepancy < 10
+        assert Decimal(signed["fee"]) == fee
```

Comment:
```suggestion
        assert_equal(Decimal(signed["fee"]), fee)
```

To get debug information on failure